src/Microsoft.Sbom.Api/Manifest/BaseManifestConfigHandler.cs

```diff
@@ -63,4 +61,39 @@ protected ISbomConfig CreateSbomConfig()
             Recorder = new SbomPackageDetailsRecorder()
         };
     }
+
+    public virtual bool TryGetManifestConfig(out ISbomConfig sbomConfig)
+    {
+        sbomConfig = CreateSbomConfig();
+
+        // For generation the default behavior is to always return true
+        // as we generate all the current formats of SBOM. Only override if the -mi
+        // argument is specified.
+        if (configuration.ManifestToolAction == ManifestToolActions.Generate)
+        {
+            if (configuration.ManifestInfo?.Value != null
+                && !configuration.ManifestInfo.Value.Contains(Constants.SPDX30ManifestInfo))
```

Comment:
You're using `Constants.SPDX30ManifestInfo` for both 22 and 30. That seems incorrect. I'd suggest making this a (non-virtual) protected method that takes a `ManifestInfo` as a parameter, then you implement the interfaces in the children classes as follows:

```
public override bool TryGetManifestConfig(out ISbomConfig sbomConfig)
{
    return TryGetManifestConfig(sbomConfig, Constants.SPDX22ManifestInfo);
}
```

and 
```
public override bool TryGetManifestConfig(out ISbomConfig sbomConfig)
{
    return TryGetManifestConfig(sbomConfig, Constants.SPDX22ManifestInfo);
}
```

Reply:
This has since been updated, it should support all valid manifestInfo's. This also removes the need for the hacky way I went about manually adding manifestInfo's in the SBOMGenerationWorkflow to make UTs work.